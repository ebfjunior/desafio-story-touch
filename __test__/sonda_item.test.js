import React from 'react';
import {Provider} from 'react-redux';
import configureMockStore from 'redux-mock-store';
import SondaItem from '../src/components/sonda_item';

import Sonda from '../src/business/sonda.js';

it('Deve renderizar corretamente', () => {
  const wrapper = mount(<Provider store={configureMockStore()()}><SondaItem/></Provider>);

  expect(wrapper).toMatchSnapshot();
});

it('Deve preencher os campos de acordo com a sonda atrelada ao componente', () => {
  const sonda = new Sonda({x: 5, y: 5, direcao: 'E', comandos: 'MMM'});
  const wrapper = mount(<Provider store={configureMockStore()()}><SondaItem sonda={sonda}/></Provider>);

  expect(wrapper.find('.x-input').props().value).toBe(5);
  expect(wrapper.find('.y-input').props().value).toBe(5);
  expect(wrapper.find('.direcao-input').props().value).toBe('E');
  expect(wrapper.find('.comandos-input').props().value).toBe('MMM');

});

it('Se existir uma sonda atrelada ao componente, o botão de adicionar não deve aparecer', () => {
  const sonda = new Sonda({x: 5, y: 5, direcao: 'E', comandos: 'MMM'});
  const wrapper = mount(<Provider store={configureMockStore()()}><SondaItem sonda={sonda}/></Provider>);

  expect(wrapper.find('button').props().style.display).toBe("none");
});

it('Se existir uma sonda atrelada ao componente, os campos de texto devem ficar desabilitados', () => {
  const sonda = new Sonda({x: 5, y: 5, direcao: 'E', comandos: 'MMM'});
  const wrapper = mount(<Provider store={configureMockStore()()}><SondaItem sonda={sonda}/></Provider>);

  expect(wrapper.find('.x-input').props().disabled).toBe(true);
  expect(wrapper.find('.y-input').props().disabled).toBe(true);
  expect(wrapper.find('.direcao-input').props().disabled).toBe(true);
  expect(wrapper.find('.comandos-input').props().disabled).toBe(true);
});

it('Se não existir uma sonda atrelada ao componente, o botão de adicionar deve aparecer', () => {
  const wrapper = mount(<Provider store={configureMockStore()()}><SondaItem/></Provider>);

  expect(wrapper.find('button').props().style.display).toBe("inline");
});

it('Deve exibir mensagem de erro quando nenhum campo é preenchido', () => {
  const wrapper = mount(<Provider store={configureMockStore()({config: {width: 5, height: 5}})}><SondaItem/></Provider>);
    wrapper.find('button').simulate('click');
    expect(wrapper.find('.errors').length).toBe(4);
  
});