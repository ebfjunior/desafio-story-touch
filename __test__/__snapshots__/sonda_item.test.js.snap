// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Deve renderizar corretamente 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(SondaItem)>
    <SondaItem
      adicionarSonda={[Function]}
    >
      <div
        className="form-group form-inline sonda-item"
        style={
          Object {
            "display": "block",
          }
        }
      >
        <input
          className="x-input form-control"
          disabled={false}
          name="x"
          onChange={[Function]}
          placeholder="Coordenada X"
          type="number"
          value=""
        />
        <input
          className="y-input form-control"
          disabled={false}
          name="y"
          onChange={[Function]}
          placeholder="Coordenada Y"
          type="number"
          value=""
        />
        <input
          className="direcao-input form-control"
          disabled={false}
          maxLength="1"
          name="direcao"
          onChange={[Function]}
          placeholder="Direção (N, S, E, W)"
          type="text"
          value=""
        />
        <input
          className="comandos-input form-control"
          disabled={false}
          name="comandos"
          onChange={[Function]}
          placeholder="L: Esquerda | R: Direita | M: Frente"
          size="40"
          type="text"
          value=""
        />
        <button
          className="btn btn-primary"
          onClick={[Function]}
          style={
            Object {
              "display": "inline",
            }
          }
          type="button"
        >
          Adicionar
        </button>
      </div>
    </SondaItem>
  </Connect(SondaItem)>
</Provider>
`;
